### PR TITLE
[ASCII-1502] Split telemetry component into component and implementation

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Working on the removing telemetry dependency from the serverless agent. While figuring out the best way to handle that I realized the telemetry component is using the old file hierarchy. This PR changes that 😄 

- I created the `telemetryimpl` package 
- Updated the code that used the old package 
- Update the tasks files 

### Motivation

With these changes I'm hoping I will be able to remove the Prometheus dependency with more ease. 

### Describe how to test/QA your changes

This PR is just moving code around so there is no behaviour change to tests 😄

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: comp/core/telemetry/component_mock.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
```
Comment (ogaca-dd) [inline comp/core/telemetry/component_mock.go:5]: Can you add `//go:build test` ?

File: comp/core/telemetry/telemetryimpl/telemetry.go [hunk 2]
```diff
@@ -3,14 +3,18 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2023-present Datadog, Inc.
 
-package telemetry
+// Package telemetryimpl implements the telemtry component interface.
```
Comment (ogaca-dd) [inline comp/core/telemetry/telemetryimpl/telemetry.go:6]: `telem*e*try` instead of `telemtry`

**Review by davidor — APPROVED** [review]
:+1: for container-integrations files

**Review by remeh — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by carlosroman — APPROVED** [review]

**Review by pducolin — APPROVED** [review]
Ok for agent-developer-tools owned files!

**Review by pgimalac — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]

**Review by dinooliva — APPROVED** [review]

**Review by daniel-taf — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Pull request merged**